### PR TITLE
    Update android build tools

### PR DESCRIPTION
**Pull request by alxndrsn**
This commit updates gradle and travis config, and also fixes related errors, including some fixes backported from upstream:

* https://github.com/opendatakit/collect/pull/523
* https://github.com/opendatakit/collect/issues/1316

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,18 @@ branches:
       - /^v.*/
 android:
   components:
-  - build-tools-22.0.1
-  - android-22
+    # some travis/android magic from https://docs.travis-ci.com/user/languages/android/
+    - tools
+    - platform-tools
+    - tools
```
Comment (SCdF) [inline .travis.yml]: Surely you don't mean to have `tools` twice? There is nothing at that URL that implies you need to do that, unless I am missing it?

File: AndroidManifest.xml [hunk 2]
```diff
@@ -107,16 +107,16 @@
...
         <activity
             android:name="org.odk.collect.android.activities.InstanceChooserList"
-            android:label="@string/app_name" />
-
-        <intent-filter>
-            <action android:name="android.intent.action.VIEW" />
-            <action android:name="android.intent.action.EDIT" />
+            android:label="@string/app_name">
+	    <intent-filter>
+		<action android:name="android.intent.action.VIEW" />
```
Comment (SCdF) [inline AndroidManifest.xml]: Did you mean to move the whitespace around like this? Seems wrong?

File: gradlew.bat [hunk 3]
```diff
@@ -46,7 +46,7 @@ echo location of your Java installation.
 goto fail
 
 :init
-@rem Get command-line arguments, handling Windowz variants
+@rem Get command-line arguments, handling Windows variants
```
Comment (SCdF) [inline gradlew.bat:49]: Do we even support windows building this anymore?

**Review by SCdF — CHANGES_REQUESTED** [review]
I have no idea what all of this does, but I imagine if it works it's proof enough.

Just one maybe wrong thing in travis and some whitespace breakage, then all good.

File: .travis.yml (hunk 1 above)
Comment (alxndrsn) [reply .travis.yml]: I will test this, but I think it's required.

File: AndroidManifest.xml (hunk 2 above)
Comment (alxndrsn) [reply AndroidManifest.xml]: Good spot, will fix.

File: .travis.yml (hunk 1 above)
Comment (alxndrsn) [reply .travis.yml]: Implied that it's required here: https://github.com/travis-ci/travis-ci/issues/6040#issuecomment-219367943; let's see what happens with these builds:
* https://travis-ci.org/medic/medic-collect/builds/475376913
* https://travis-ci.org/medic/medic-collect/builds/475376846

Comment (alxndrsn) [reply .travis.yml]: If it's required, I'll reference the Travis issue in a comment in the yaml file.

Comment (alxndrsn) [reply .travis.yml]: Finally, I've got rid of both `tools` references _and_ the `platform-tools`.

**Review by SCdF — APPROVED** [review]
LGTM!

**Pull request merged**